google-login-plugin/src/com/google/gct/login/stats/UsageTrackerService.java

```diff
@@ -15,52 +15,28 @@
  */
 package com.google.gct.login.stats;
 
-import com.intellij.openapi.util.KeyedExtensionCollector;
-import com.intellij.util.PlatformUtils;
+import com.intellij.openapi.components.ServiceManager;
 
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 /**
- * Usage Tracker Service for obtaining extensions declared in plugin.xml with {@link
- * UsageTrackerExtensionPointBean}
+ * Usage Tracker Service for obtaining UsageTracking implementations
  */
-public final class UsageTrackerService {
+public abstract class UsageTrackerService {
 
-  private static final KeyedExtensionCollector<UsageTracker, String> COLLECTOR =
-      new KeyedExtensionCollector<UsageTracker, String>(
-          UsageTrackerExtensionPointBean.EP_NAME.getName());
-  private static final UsageTrackerManager usageTrackerManager = UsageTrackerManager.getInstance();
-
-  /**
-   * When using the usage tracker, do NOT include any information that can identify the user
-   *
-   * @return the usage tracker for the current platform
-   */
   @NotNull
   public static UsageTracker getInstance() {
-    String key = PlatformUtils.getPlatformPrefix();
-    return getInstance(key);
+    return ServiceManager.getService(UsageTrackerService.class).getTracker();
```

Comment:
Well the reason we made this a service is because intellij provides a mechanism to mock these services out. Presumably this should make testing easier.